### PR TITLE
Add go.mod and go.sum files to support Go modules.

### PR DESCRIPTION
**Pull request by creachadair**
This commit contains no functional changes; it's just adding config files for
the Go modules facility. Ideally this commit should also be accompanied by a
new version tag, since the current latest tag is v1.0.0 from Nov. 2017, and
several optimizations have been added since then.

### CODE REVIEW

**Comment by creachadair** [comment]
Clearly the CI is in some way discontent with this change, I'm trying to work out why.

**Comment by creachadair** [comment]
Ah, I think I see the problem. The last successful CI build was 6 months ago on #94, and since then the  [errcheck](https://github.com/kisielk/errcheck) package has updated to depend on Go ≥ 1.9. The Travis config for 1.8 will therefore no longer work at all, irrespective of modules.

**Comment by creachadair** [comment]
Indeed, CI fails at HEAD: https://travis-ci.org/sergi/go-diff/jobs/488608466. So this PR didn't cause any new problems, but there are some existing ones that will need fixed. I'll look into that separately.

**Comment by creachadair** [comment]
Ok, I was able to fix the import path issue for `golang.org/x/lint` for 1.9, but 1.8 remains broken because the API is no longer compatible. Specifically, 1.8 predates type alias support.

**Comment by creachadair** [comment]
I suggest dropping support for Go 1.8 entirely. Even though this repo only depends on golint as a verification tool, the dependency compatibility problem runs deeper. To address it you'll need to either install a newer version of Go to build golint, or find an older binary release to depend on (which golint does not itself provide).

**Comment by creachadair** [comment]
After that somewhat noisy debugging, I think this is ready for a review.

**Comment by creachadair** [comment]
I'm not sure who maintains the package, but if someone would be willing to take a look and let me know if this seems reasonable, I'd be grateful! I'm happy to address any concerns promptly.

**Comment by creachadair** [comment]
Ping! Is someone available to take a look and let me know whether you consider this a reasonable change? Thanks.

**Comment by creachadair** [comment]
Ping! @sergi do you have any thoughts about this proposed change?

**Comment by creachadair** [comment]
Ping! Is anyone maintaining this package, and willing to offer comment about this change?

**Comment by creachadair** [comment]
Ping! Can someone please take a look at this change? Even if you have concerns or objections, it would be helpful for us to know what they are. Thanks.

**Comment by creachadair** [comment]
Ping! Is someone available to take a look and let me know whether you consider this a reasonable change? Thanks.

**Comment by creachadair** [comment]
Ping! Can you please take a look when you have time?

**Comment by paper777** [comment]
@sergi PING
PTAL
Thanks!

**Comment by creachadair** [comment]
Ping, @sergi. Please have a look when you can, thanks.

**Comment by creachadair** [comment]
@sergi Please take a look when you are able.

**Comment by creachadair** [comment]
Hi, @sergi, I'd be grateful for a review when you are able. Thanks!

**Comment by creachadair** [comment]
Greetings, @sergi! Please take a look when you have some time.

**Comment by creachadair** [comment]
Dear @sergi, can you please have a look at this PR when you have time? Thanks!

**Comment by edocevol** [comment]
@sergi  Please review PR~~~

**Comment by creachadair** [comment]
Hello, @sergi, when you have some available bandwidth, I'd be very grateful for a review here. If you have any questions or concerns about the change, I'll do my best to address them. Or if you disagree with the change entirely, it would be helpful to know why. Thanks!

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,9 @@ os:
   - osx
 
 go:
-  - 1.8.x
   - 1.9.x
+  - 1.10.x
+  - 1.11.x
```
[the remark this answers is not in the document]
Comment (creachadair) [reply .travis.yml]: Good idea—done!

**Comment by creachadair** [comment]
Thank you!

**Comment by creachadair** [comment]
Would you like me to squash the commits before merge?

**Comment by creachadair** [comment]
I went ahead and pre-squashed the commits so the merge will be clean once CI completes.

**Comment by creachadair** [comment]
Hi, @sergi, I think this is ready for a merge now (I don't have write access). Thanks!

**Comment by creachadair** [comment]
Hi, @sergi, I've updated the Travis CI config to include Go 1.13 now that it has been released, and added the new Go tool version tag to `go.mod`. Once CI clears this should be ready to merge at your convenience.

**Pull request merged**